core/app/components/forms/elements/ContextEditorElement.tsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import type { Node } from "prosemirror-model";
+
+import { useMemo } from "react";
+import { Value } from "@sinclair/typebox/value";
+import { useFormContext } from "react-hook-form";
+import { richTextInputConfigSchema } from "schemas";
+
+import { FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from "ui/form";
+
+import type { ElementProps } from "../types";
+import { ContextEditorClient } from "../../ContextEditorClient";
+import { useFormElementToggleContext } from "../FormElementToggleContext";
+
+const EditorFormElement = ({
+	label,
+	help,
+	onChange,
+	initialValue,
+}: {
+	label: string;
+	help?: string;
+	onChange: (state: any) => void;
+	initialValue?: Node;
+}) => {
+	const memoEditor = useMemo(() => {
+		return (
+			<ContextEditorClient
+				pubs={[]}
+				pubTypes={[]}
+				onChange={onChange}
+				initialDoc={initialValue}
+			/>
+		);
+	}, [onChange]);
+
+	return (
+		<FormItem>
+			<FormLabel className="flex">{label}</FormLabel>
+			<div className="w-full">
+				<FormControl>{memoEditor}</FormControl>
+			</div>
+			<FormDescription>{help}</FormDescription>
+			<FormMessage />
+		</FormItem>
+	);
+};
+
+export const ContextEditorElement = ({ name, config }: ElementProps) => {
+	const { control } = useFormContext();
+	const formElementToggle = useFormElementToggleContext();
+	const isEnabled = formElementToggle.isEnabled(name);
+
+	Value.Default(richTextInputConfigSchema, config);
+	if (!Value.Check(richTextInputConfigSchema, config)) {
+		return null;
+	}
+
+	return (
+		<FormField
+			control={control}
+			name={name}
+			render={({ field }) => {
+				return (
+					<EditorFormElement
+						label={config.label ?? name}
+						help={config.help}
+						// I think this should be state.doc.toJSON() but that causes infinite rendering :(
+						onChange={(state) => field.onChange(state.doc)}
```

Comment:
I could use some help getting this to work! not sure if there's anything that can be done external to the context editor to get this to work. but this basically makes it so that someone else needs to convert to JSON on submit. I added that workaround on the form flow, but not yet for the pub create/update flow, since ideally this part would work

Reply:
I think maybe we don't want this for the Editor, I think that's a huge performance hit for the user to have to serialize the entire tree every render. Instead we probably only should serialize this on "save" (don't know what that looks like) or submit, mayyybe on "blur". 

I'm not really sure what that looks like here. Maybe what you have for the Form flow is fine.

Don't really know whether we need to update the state of the of the form anytime something is changed in the editor really. Is it possible to just leave this `onChange` out and only get the state of the field on submit, or is that not how `react-hook-form` works?

As for the infinite rerendering: I'm guessing it has something to do with the fact you would be creating a new object every time `toJSON()` is called. But still I don't know what makes that happen, maybe the `initialValue` is used in the editor as the value as well?

Reply:
ah got it! that makes sense, and I imagine we'll have to do some other processing with the editor data on save anyway (like modifying the related pubs and all that). I'll look into if `react-hook-form` a bit more, but I think it would be okay to have all the processing on the `onSubmit`

Reply:
btw i think the problem is in here: https://github.com/pubpub/platform/blob/36088913aeec23b447a8fd9d82ef9a00ce02ad16/packages/context-editor/src/ContextEditor.tsx#L136-L139

@isTravis  also left a comment about this, it seems he's doing custom there.
It does seem to have to do with creating a new object, the same thing happens when you put `{}` instead of `state.doc.toJSON()` in the `field.onChange`. I tried for a while to understand it, but I don't really get it tbh!